### PR TITLE
PIM-10002: Pull-up 5.0 -> Master

### PR DESCRIPTION
**Pull request by LaurentPetard**

### CODE REVIEW

**Review by ValentinMumble — APPROVED** [review]
GTM after handling master TODOs 😉

File: src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php [hunk 1]
```diff
@@ -33,22 +37,35 @@ class ChannelLocaleSubscriber implements EventSubscriberInterface
...
+
+    /** @var IdentifiableObjectRepositoryInterface | null */
+    private $jobInstanceRepository;
+
     /**
-     * @param LocaleRepositoryInterface $repository
-     * @param BulkSaverInterface        $saver
-     * @param CommandLauncher           $commandLauncher
-     * @param TokenStorageInterface     $tokenStorage
+     * TODO On merge master remove $commandLauncher
```
Comment (ValentinMumble) [inline src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php:50]: 👀

File: tests/back/Channel/Specification/Bundle/EventListener/ChannelLocaleSubscriberSpec.php [hunk 2]
```diff
@@ -68,8 +83,91 @@ function it_does_not_upadte_channel_on_non_channels($saver, GenericEvent $event,
...
+        $localeEs->addChannel($channel)->shouldBeCalled();
+
+        $saver->saveAll([$localeFr, $localeEs])->shouldBeCalled();
+
+        $jobLauncher->launch(Argument::cetera())->shouldNotBeCalled();
+
+        $this->updateChannel($event);
+    }
+
+    // TODO Remove this test on merge master
```
Comment (rfauglas) [inline tests/back/Channel/Specification/Bundle/EventListener/ChannelLocaleSubscriberSpec.php:164]: Remove?

File: tests/back/Channel/Specification/Bundle/EventListener/ChannelLocaleSubscriberSpec.php [hunk 3]
```diff
@@ -102,13 +200,21 @@ function it_updates_channel_when_a_locale_has_been_removed(
...
+            $saver,
+            $commandLauncher,
+            $tokenStorage,
+        );
+
         $this->updateChannel($event);
     }
 
-    function it_updates_channel_without_removed_locale(
+    // TODO Remove this test on merge master
```
Comment (rfauglas) [inline tests/back/Channel/Specification/Bundle/EventListener/ChannelLocaleSubscriberSpec.php:213]: Same as above?

**Review by rfauglas — CHANGES_REQUESTED** [review]
handle TODOs ;-)

**Review by rfauglas — APPROVED** [review]

**Pull request merged**